### ISSUE TITLE
bootBuildImage task hangs while saving docker image if using azul-zulu

### ISSUE THREAD

**Issue by xuan-nguyen-swe** [paketo-buildpacks/spring-boot#312]
`bootBuildImage` task hangs while saving docker image if using gcr.io/paketo-buildpacks/azul-zulu

```
tasks {
  bootBuildImage {
    environment.set(
      mapOf(
        "BP_JVM_VERSION" to "17"
      )
    )

    // https://paketo.io/docs/howto/java/#use-an-alternative-jvm
    buildpacks.set(
      listOf(
        "paketo-buildpacks/ca-certificates",
        "gcr.io/paketo-buildpacks/azul-zulu",
        "paketo-buildpacks/java",
      )
    )
  }
}
```

**Comment by xuan-nguyen-swe** [comment]
@scottfrederick Thanks for your prompt reply.
I tried again with a clean project, and it works. So, it must be smt else incompatible in my original project.
But it's strange. It works if I don't customize configuration of `buildpacks`.
Let's me check again and give you more information if I can find smt.

**Comment by dmikusa** [comment]
Could it be a network issue? When you don't set any buildpacks, it'll load those from the builder which is just a single image that's needed. 

When you put in the buildpacks, more network requests happen. It needs to go fetch those images. Maybe one of those requests is hanging?

**Comment by xuan-nguyen-swe** [comment]
> Could it be a network issue?

No, I don't think so. It hangs at the last step to save the built image as local docker image.
I am trying to reduce my project until I can reproduce with the simplest gradle project then I will post it again here.

**Comment by xuan-nguyen-swe** [comment]
I know how to reproduce the bug now.
Actually, it's not related to buildpacks configuration because I can reproduce without configuring bootBuildImage task.

File `buildSrc/build.gradle.kts`
```
plugins {
  `kotlin-dsl`
}

repositories {
  mavenCentral()
  gradlePluginPortal()
}

// Add Plugin Marker Artifacts dependencies, so convention plugins could resolve plugins
dependencies {
  val kotlinVersion = "1.8.0"
  implementation(kotlin("gradle-plugin", kotlinVersion))
  implementation(kotlin("allopen", kotlinVersion))
  implementation(kotlin("reflect", kotlinVersion))
  implementation(kotlin("stdlib-jdk8", kotlinVersion))

  implementation("org.springframework.boot:spring-boot-gradle-plugin:3.0.2")
}
```

File `build.gradle.kts`
```
plugins {
	kotlin("jvm")
	kotlin("plugin.spring")

	id("org.springframework.boot")
	id("io.spring.dependency-management")
}

group = "com.example"
version = "0.0.1-SNAPSHOT"
java.sourceCompatibility = JavaVersion.VERSION_17

repositories {
	mavenCentral()
}

dependencies {
	implementation(kotlin("reflect"))
	implementation("org.springframework.boot:spring-boot-starter")
	testImplementation("org.springframework.boot:spring-boot-starter-test")
}

tasks {
  test {
    useJUnitPlatform()
  }

  compileKotlin {
    kotlinOptions {
      // Enable strict mode for null-safety
      // Many libraries, such as Spring, encourage this configuration
      freeCompilerArgs = listOf("-Xjsr305=strict")
      jvmTarget = project.java.targetCompatibility.majorVersion
    }
  }
}
```

Now, run gradle tasks: `clean bootBuildImage`.
If success, update `version = "0.0.1-SNAPSHOT"` to next version until docker is frozen.
After 4 times update the version and run `clean bootBuildImage`, my docker hangs at 0.0.4-SNAPSHOT.

[log-successful.txt](https://github.com/paketo-buildpacks/spring-boot/files/10472409/log-successful.txt)
[log-frozen.txt](https://github.com/paketo-buildpacks/spring-boot/files/10472408/log-frozen.txt)

**Comment by xuan-nguyen-swe** [comment]
Oh, the same behavior even if I remove kotlin completely.

`buildSrc/build.gradle.kts`
```
plugins {
  `kotlin-dsl`
}

repositories {
  gradlePluginPortal()
}

// Add Plugin Marker Artifacts dependencies, so convention plugins could resolve plugins
dependencies {
  implementation("org.springframework.boot:spring-boot-gradle-plugin:3.0.2")
}
```

```
plugins {
  java
	id("org.springframework.boot")
	id("io.spring.dependency-management")
}

group = "com.example"
version = "0.0.3-SNAPSHOT"
java.sourceCompatibility = JavaVersion.VERSION_17

repositories {
	mavenCentral()
}

dependencies {
	implementation("org.springframework.boot:spring-boot-starter")
	testImplementation("org.springframework.boot:spring-boot-starter-test")
}

tasks {
  test {
    useJUnitPlatform()
  }
}
```

**Comment by xuan-nguyen-swe** [comment]
@scottfrederick @dmikusa I updated the original post with the easiest way to reproduce the bug.

**Comment by fabioportieri** [comment]
i have the exact same issue, just opened a ticket: https://github.com/spring-projects/spring-boot/issues/34026
i managed to reproduce it with a brand new application, it happens in distinct windows machine (win10 and win11)
i'll see if i can reproduce it on my linux machine too

**Comment by dmikusa** [comment]
@fabioportieri what are you using for Docker (software/version)? When it's saving and your Docker becomes unresponsive, can you do other tasks with Docker? Like `docker ps`?

**Comment by scottfrederick** [comment]
@xuan-nguyen-swe and @fabioportieri Can you try to reproduce the problem using the `pack` CLI as [requested earlier](https://github.com/paketo-buildpacks/spring-boot/issues/312#issuecomment-1398741993)? This would be very helpful for us to understand if the problem has anything to do with the Spring Boot plugins.

**Comment by xuan-nguyen-swe** [comment]
> @xuan-nguyen-swe and @fabioportieri Can you try to reproduce the problem using the `pack` CLI as [requested earlier](https://github.com/paketo-buildpacks/spring-boot/issues/312#issuecomment-1398741993)? This would be very helpful for us to understand if the problem has anything to do with the Spring Boot plugins. 

I will try to use cli and tell me result later. I am totally new to buildpacks so I need to check how to setup it.

**Comment by fabioportieri** [comment]
docker becomes completely unresponsive even from commandline, the only thing i can do is to restart the machine, the issue has been reproduced in two different windows machines (win10 and win11), as i said i will try to reproduce in a native ubuntu installation, regarding the pack cli if you kindly provide me with the correct command i'll execute it straight away

i tried this:
```
pack config default-builder paketobuildpacks/builder:base 
```
```
pack build -e BP_JVM_VERSION=11 --buildpack "paketo-buildpacks/ca-certificates" --buildpack "paketo-buildpacks/bellsoft-liberica" --buildpack "paketo-buildpacks/syft" --buildpack "paketo-buildpacks/executable-jar" --buildpack "paketo-buildpacks/dist-zip" --buildpack "paketo-buildpacks/spring-boot" -p target/dataflowmanager-1.0-SNAPSHOT.jar localhost:5000/dataflow-manager:1.0-SNAPSHOT
```

but ended up with this error:
```
Digest: sha256:960420f8e05845e9acbc4e04de77dc852ca0823bbdf640d9e9f4131a89f8d1fe
Status: Downloaded newer image for paketobuildpacks/run:base-cnb
===> ANALYZING
===> DETECTING
ERROR: No buildpack groups passed detection.
ERROR: Please check that you are running against the correct path.
ERROR: failed to detect: no buildpacks participating
ERROR: failed to build: executing lifecycle: failed with status code: 20
```

**Comment by scottfrederick** [comment]
> regarding the pack cli i tried and i had the same issue

Thanks for trying that @fabioportieri. That confirms that the problem is not related to the Spring Boot plugins or the `pack` CLI directly, and allows us to focus the troubleshooting at the layers shared by both tools.

**Comment by xuan-nguyen-swe** [comment]
Thanks @fabioportieri for testing the issue with pack cli.
I think it's quite clear now, so I don't need to test pack cli again.

**Comment by dmikusa** [comment]
@fabioportieri what are you using for Docker (software/version)? I assume Docker Desktop for Windows, but if you could confirm & let us know the specific version as well. If it's not the latest version, please update. Thanks

**Comment by fabioportieri** [comment]
yes, docker desktop, updated to latest version v4.16.3, for windows


this is my docker info

Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.10.0)
  compose: Docker Compose (Docker Inc., v2.15.1)
  dev: Docker Dev Environments (Docker Inc., v0.0.5)
  extension: Manages Docker extensions (Docker Inc., v0.2.17)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.23.0)

Server:
 Containers: 37
  Running: 13
  Paused: 0
  Stopped: 24
 Images: 14
 Server Version: 20.10.22
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runtime.v1.linux runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9ba4b250366a5ddde94bb7c9d1def331423aa323
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.15.79.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 20
 Total Memory: 15.47GiB
 Name: docker-desktop
 ID: 32I6:KQAG:IIZJ:G5OH:ASNH:JFHJ:4RAS:KXEY:LG2C:HDZN:7VWJ:AFZO
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false

**Comment by anthonydahanne** [comment]
Hello,
I've tried to reproduce the issue in a public CI, for everybody to see how to reproduce.

I've successfully built on a Windows 10 machine, 

```
Nom du système d’exploitation:              Microsoft Windows 10 Professionnel
Version du système:                         10.0.19044 N/A version 19044
Type du système:                            x64-based PC
Processeur(s):                              1 processeur(s) installé(s).
                                            [01] : Intel64 Family 6 Model 142 Stepping 9 GenuineIntel ~2703 MHz
```

with that `~/.wslconfig`

```
[wsl2]
#kernel=C:\\temp\\myCustomKernel
memory=4GB # Limits VM memory in WSL 2 to 4 GB
processors=2 # Makes the WSL 2 VM use two virtual processors
```

and Docker For Windows v20.10.8 with `Enable integration with my default WSL distro`

* a JVM based image from a simple Spring Boot project using Pack CLI
* a JVM based image from a simple Spring Boot project using Spring Boot Maven Plugin
* a Native based image, from the same Spring boot project using Spring Boot Maven Plugin

a fair number of times now: https://github.com/anthonydahanne/spring-boot-demo/actions/workflows/windows.yml


@fabioportieri @xuan-nguyen-swe : I invite you to clone my repo and use your own machine as a runner, see if you still can reproduce your issues: the advantage of using GH actions is that it will allow us to investigate further and understand exactly under which conditions we can reproduce the issue.

Without any additional context / logs / recorded builds, I'd say the issue does not happen anymore. 

Thanks!

**Comment by Azbesciak** [comment]
I also confirm the same issue, I am using spring boot, mostly with AOT builds but I also reproduced that with non-aot build.
I have that since January, 70% of the build crashed docker, the only solution is to restart the computer (I even cannot kill docker); comp works normally, but I will not run any docker image, and that one is hanging.

**Comment by anthonydahanne** [comment]
Can you please share a project that makes docker hang, along with the command you used?
In a previous comment,  I stated I built dozen of times a project on Windows,  without any issues.
So every detail helps here.
Thanks
